lib/postgrex/protocol.ex

```diff
@@ -141,16 +146,30 @@ defmodule Postgrex.Protocol do
     end
   end
 
+  defp connect_endpoints([{_host, _port} | _] = endpoints, sock_opts, timeout, s, status, previous_errors) do
+    endpoints = Enum.map(endpoints, fn {host, port} -> {host, port, []} end)
+    connect_endpoints(endpoints, sock_opts, timeout, s, status, previous_errors)
+  end
+
   defp connect_endpoints(
-         [{host, port} | remaining_endpoints],
+         [{host, port, extra_ssl_opts} | remaining_endpoints],
```

Comment:
Instead of this being `ssl_opts`, I would prefer if this was `extra_sock_opts`. You can then call `Config.Reader.merge(sock_opts, extra_sock_opts)` to merge them!